src/serializers/comparisons.js

```diff
@@ -63,10 +62,10 @@ var oppositeTypeMap = {
 	NotIn: { Type: is.In, prop: "values" }
 };
 
-hydrateMap["$not"] = function(value, unknownHydrator) {
+hydrateMap.$not = function(value, unknownHydrator) {
 	// Many nots can be hydrated to their opposite.
-	var hydratedValue = hydrateValue(value["$not"], unknownHydrator);
-	var typeName = hydratedValue.constructor.name;
+	var hydratedValue = hydrateValue(value.$not, unknownHydrator);
+	var typeName = hydratedValue.constructor.name || hydratedValue.constructor.toString().match(/^\s*function\s*(\S*)\s*\(/)[1];
```

Comment:
This is a hack to get IE 11 working because `constructor.name` isn’t available.

I think we should probably fix https://github.com/canjs/can-reflect/issues/167 and add a `getConstructorName` method to can-reflect that returns just the constructor name (instead of something like `ConstructorName{}`, which is what `getName` returns).

Otherwise, we could maybe refactor this code to not rely on the constructor name… I think that would be a lot more involved though because there’s a lot of functions involved with the `hydrateValue()` call.

Reply:
Can you explain the problem with `getName`? I'm not sure I follow.

Reply:
I just realized that you can pass in a constructor function to `getName` and it’ll return just the name, not something like `ConstructorName{}`. Scratch what I said about adding a separate `getConstructorName` method.